File: calculadora.py
import math

def menu():
    print("\n=== Calculadora Científica ===")
    print("1. Soma")
    print("2. Subtração")
    print("3. Multiplicação")
    print("4. Divisão")
    print("5. Potência")
    print("6. Raiz Quadrada")
    print("7. Fatorial")
    print("0. Sair")

def calcular():
    while True:
        menu()
        opcao = input("Escolha uma operação (0-7): ")
        
        if opcao == "0":
            print("Até logo!")
            break
        
        if opcao in ["1", "2", "3", "4", "5"]:
            num1 = float(input("Digite o primeiro número: "))
            num2 = float(input("Digite o segundo número: "))
        
        if opcao == "1":
            print(f"Resultado: {num1 + num2}")
        elif opcao == "2":
            print(f"Resultado: {num1 - num2}")
        elif opcao == "3":
            print(f"Resultado: {num1 * num2}")
        elif opcao == "4":
            print(f"Resultado: {num1 / num2}")
        elif opcao == "5":
            print(f"Resultado: {num1 ** num2}")
        elif opcao == "6":
            num = float(input("Digite um número: "))
            print(f"Resultado: {math.sqrt(num)}")
        elif opcao == "7":
            num = int(input("Digite um número inteiro: "))
            print(f"Resultado: {math.factorial(num)}")
        else:
            print("Opção inválida!")

if __name__ == "__main__":
    calcular()

File: calculadora.js
// Lógica da calculadora (versão corrigida)
function calcular(opcao, valores) {
    try {
        if (opcao >= "1" && opcao <= "5") {
            const num1 = parseFloat(valores.num1);
            const num2 = parseFloat(valores.num2);
            
            if (isNaN(num1) || isNaN(num2)) {
                throw new Error("Valores inválidos!");
            }

            switch(opcao) {
                case "1": return num1 + num2;
                case "2": return num1 - num2;
                case "3": return num1 * num2;
                case "4": 
                    if (num2 === 0) throw new Error("Divisão por zero!");
                    return num1 / num2;
                case "5": return Math.pow(num1, num2);
                default: throw new Error("Operação inválida!");
            }
        } 
        else if (opcao === "6") {
            const num = parseFloat(valores.num);
            if (isNaN(num)) throw new Error("Valor inválido!");
            if (num < 0) throw new Error("Não existe raiz de número negativo!");
            return Math.sqrt(num);
        } 
        else if (opcao === "7") {
            const num = parseInt(valores.num);
            if (isNaN(num)) throw new Error("Valor inválido!");
            if (num < 0) throw new Error("Não existe fatorial de número negativo!");
            if (num > 170) throw new Error("Número muito grande para cálculo de fatorial!");
            
            let resultado = 1;
            for (let i = 2; i <= num; i++) {
                resultado *= i;
            }
            return resultado;
        }
        else {
            throw new Error("Operação inválida!");
        }
    } catch (error) {
        throw error;
    }
}